torch/_subclasses/fake_impls.py

```diff
@@ -913,14 +913,12 @@ def infer_size(a, b):
         # were not the case, we'd need to write this using torch.sym_or() or
         # something like that).
         torch._check(
-            guard_size_oblivious(sizeA == 1)
-            or guard_size_oblivious(sizeB == 1)
-            or sizeA == sizeB,
+            guard_or_false(sizeA == 1) or guard_or_false(sizeB == 1) or sizeA == sizeB,
```

Comment:
you can land it as for the future i might revisit this to make the code more understandable

basically first we want to check if there is broadcasting using guard_or_none.
if any of them did not return none we are done. 
if both return non i would want to add an explicit extra message to the torch_check that says we have assumed this path because both sizeA == sizeB are unbacked

not action required from you at this momment. i will file issue for this.